lib/matplotlib/quiver.py

```diff
@@ -513,58 +525,233 @@ def _init(self):
+        _api.warn_deprecated("3.9", alternative="get_V")
+        return self.get_V()
+
+    @V.setter
+    def V(self, value):
+        _api.warn_deprecated("3.9", alternative="set_V")
+        return self.set_V(value)
+
+    @property
+    def XY(self):
+        _api.warn_deprecated("3.9", alternative="get_offsets")
+        return self.get_offsets()
+
+    @XY.setter
+    def XY(self, value):
+        _api.warn_deprecated("3.9", alternative="set_offsets")
+        self.set_offsets(offsets=value)
+
+    def set_offsets(self, offsets):
+        self.set_XYUVC(X=offsets[:, 0], Y=offsets[:, 1])
+
     def get_datalim(self, transData):
         trans = self.get_transform()
         offset_trf = self.get_offset_transform()
         full_transform = (trans - transData) + (offset_trf - transData)
-        XY = full_transform.transform(self.XY)
+        XY = full_transform.transform(self.get_offsets())
         bbox = transforms.Bbox.null()
         bbox.update_from_data_xy(XY, ignore=True)
         return bbox
 
     @martist.allow_rasterization
     def draw(self, renderer):
         self._init()
-        verts = self._make_verts(self.XY, self.U, self.V, self.angles)
+        verts = self._make_verts(self.get_offsets(), self._U, self._V, self.angles)
         self.set_verts(verts, closed=False)
         super().draw(renderer)
         self.stale = False
 
-    def set_UVC(self, U, V, C=None):
+    def set_X(self, X):
+        """
+        Set positions in the horizontal direction.
+
+        Parameters
+        ----------
+        X : array-like
+            The size must the same as the existing Y.
+        """
+        self.set_XYUVC(X=X)
+
+    def get_X(self):
+        """Returns the positions in the horizontal direction."""
+        return self.get_offsets()[..., 0]
+
+    def set_Y(self, Y):
+        """
+        Set positions in the vertical direction.
+
+        Parameters
+        ----------
+        Y : array-like
+            The size must the same as the existing X.
+        """
+        self.set_XYUVC(Y=Y)
+
+    def get_Y(self):
+        """Returns the positions in the vertical direction."""
+        return self.get_offsets()[..., 1]
+
+    def set_U(self, U):
+        """
+        Set horizontal direction components.
+
+        Parameters
+        ----------
+        U : array-like
+            The size must the same as the existing X, Y, V or be one.
+        """
+        self.set_XYUVC(U=U)
+
+    def get_U(self):
+        """Returns the horizontal direction components."""
+        return self._U
+
+    def set_V(self, V):
+        """
+        Set vertical direction components.
+
+        Parameters
+        ----------
+        V : array-like
+            The size must the same as the existing X, Y, U or be one.
+        """
+        self.set_XYUVC(V=V)
+
+    def get_V(self):
+        """Returns the vertical direction components."""
+        return self._V
+
+    def set_C(self, C):
+        """
+        Set the arrow colors.
```

Comment:
Not really the colour; it's the data that's mapped to colours. One has to be careful about this, because there is also a `color` argument to `quiver`, which this is not.